### PR TITLE
Fix division-by-zero in lr_scheduler parameter validation (#177833)

### PR DESCRIPTION
**Pull request by jnsmrll**
**Summary**
Several `lr_scheduler` classes silently accept zero values for parameters, then raise a `ZeroDivisionError` at runtime. This PR adds validation so users get a clear error message:

- **StepLR**: reject `step_size == 0`
- **CosineAnnealingLR**: reject `T_max == 0`
- **CyclicLR**: reject `step_size_up == 0`
- **ConstantLR**: tighten existing `factor` validation from `< 0` to `<= 0`, since `factor=0` also causes division by zero in `get_lr()`

**Known issues not addressed**
- The learning rate itself can still go negative, which doesn't produce meaningful behavior. Restricting this can be done as a follow-up.
- Two additional `ZeroDivisionError` cases identified which are only happening in the deprecated use case: In `LinearLR` and in `PolynomialLR`: `total_iters=0` causes `ZeroDivisionError` when using the closed form `.step(epoch=..)`

**Test plan**
- Added `test_step_lr_step_size_zero`, `test_constantlr_factor_zero`, `test_cos_anneal_lr_T_max_zero`, and `test_cycle_lr_step_size_up_zero` to verify each scheduler raises `ValueError` on zero input
- Existing `lr_scheduler` tests pass: test/optim/test_lrscheduler.py

Addresses:
#177833 and #174025

### CODE REVIEW

File: test/optim/test_lrscheduler.py [hunk 1]
```diff
@@ -350,6 +350,11 @@ def test_step_lr(self):
         scheduler = StepLR(self.opt, gamma=0.1, step_size=3)
         self._test(scheduler, targets, epochs)
 
+    def test_step_lr_step_size_zero(self):
+        start_step_size = 0
+        with self.assertRaises(ValueError):
+            StepLR(self.opt, step_size=start_step_size)
```
Comment (janeyx99) [inline test/optim/test_lrscheduler.py]:
```suggestion
        with self.assertRaises(ValueError):
            StepLR(self.opt, step_size=0)
```

File: test/optim/test_lrscheduler.py [hunk 2]
```diff
@@ -445,6 +450,11 @@ def test_constantlr(self):
         scheduler = ConstantLR(self.opt, factor=1.0 / 2, total_iters=5)
         self._test(scheduler, targets, epochs)
 
+    def test_constantlr_factor_zero(self):
+        factor = 0
+        with self.assertRaises(ValueError):
+            ConstantLR(self.opt, factor=factor)
```
Comment (janeyx99) [inline test/optim/test_lrscheduler.py]:
```suggestion
        with self.assertRaises(ValueError):
            ConstantLR(self.opt, factor=0)
```

File: torch/optim/lr_scheduler.py [hunk 3]
```diff
@@ -1904,6 +1908,8 @@ def __init__(
 
         self.max_lrs = _format_param("max_lr", optimizer, max_lr)
 
+        if step_size_up == 0:
```
Comment (Skylion007) [inline torch/optim/lr_scheduler.py]: Ugh, why not do proper validation here with > ?

Comment (jnsmrll) [reply torch/optim/lr_scheduler.py]: Sorry, first time contributor here :)
My primary goal was to prevent `DivisionByZero` errors, but I can certainly make this clamped to `<= 0` as negative input is not logical.

File: test/optim/test_lrscheduler.py [hunk 4]
```diff
@@ -530,6 +540,11 @@ def test_cos_anneal_lr(self):
         scheduler = CosineAnnealingLR(self.opt, T_max=epochs, eta_min=eta_min)
         self._test(scheduler, targets, epochs)
 
+    def test_cos_anneal_lr_T_max_zero(self):
+        t_max = 0
```
Comment (janeyx99) [inline test/optim/test_lrscheduler.py]: same here and below, no need to create local variables here when they're used once and are named through the keywords

File: test/optim/test_lrscheduler.py [hunk 5]
```diff
@@ -350,6 +350,11 @@ def test_step_lr(self):
         scheduler = StepLR(self.opt, gamma=0.1, step_size=3)
         self._test(scheduler, targets, epochs)
 
+    @parametrize("step_size", [0, -1])
+    def test_step_lr_step_size_non_positive(self, step_size):
+        with self.assertRaises(ValueError):
```
Comment (janeyx99) [inline test/optim/test_lrscheduler.py]: You can use assertRaisesRegex to make sure the message is correct

**Review by janeyx99 — COMMENTED** [review]
@diputs-sudo does this adequately fix ur test case?

File: test/optim/test_lrscheduler.py (hunk 5 above)
Comment (jnsmrll) [reply test/optim/test_lrscheduler.py]: @janeyx99 Thanks! I added this and included the parameter names in the error message to uniquely identify them.
